src/main/java/core/basesyntax/storage/Storage.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> fruits = new HashMap<>();
+
+    public Map<String, Integer> getFruits() {
+        return fruits;
+    }
+
+    public void put(String fruit, Integer quantity) {
+        fruits.put(fruit, quantity);
+    }
+
+    public void plus(String fruit, Integer quantity) {
+        Integer initialAmount = fruits.get(fruit);
+        fruits.put(fruit, initialAmount + quantity);
+    }
+
+    public void minus(String fruit, Integer quantity) {
+        Integer initialAmount = fruits.get(fruit);
+        if (initialAmount == null) {
+            throw new RuntimeException("Can't purchase not existent fruit");
+        }
+        if (quantity > initialAmount) {
+            throw new RuntimeException("Can't remove more than there is in stock");
+        }
+        fruits.put(fruit, initialAmount - quantity);
+    }
```

Comment:
as I said earlier let's move **all** of that checks and everything out of the storage class to the handlers because it is not the responsibility of the dao, it should have just straight access to the **public** `fruits` map